Core/Object Arts/Dolphin/IDE/Base/SmalltalkWorkspace.cls

```diff
@@ -902,46 +909,20 @@ messagesForToken: anAssociation startingWith: aString maxItems: anInteger
 	"Private - Answer the set of selectors that could potentially be sent as messages following
 	the specified token."
 
-	| ignoreCase icon |
+	| selectors class ignoreCase icon |
 	ignoreCase := self isAutoCompletionCaseInsensitive.
 	icon := Message icon.
-	^((self classForToken: anAssociation)
-		ifNil: 
-			[| allSelectors |
-			allSelectors := Smalltalk developmentSystem allSelectors.
-			aString isEmpty
-				ifTrue: [allSelectors size > anInteger ifTrue: [#()] ifFalse: [allSelectors]]
-				ifFalse: 
-					[| messages |
-					messages := Set new.
-					allSelectors do: 
-							[:eachName |
-							(eachName beginsWith: aString ignoreCase: ignoreCase)
-								ifTrue: 
-									[messages add: eachName.
-									messages size > anInteger ifTrue: [^#()]]].
-					messages]]
-		ifNotNil: 
-			[:class |
-			| messages |
-			messages := Set new.
-			class withAllSuperclassesDo: (aString isEmpty
-						ifTrue: 
-							[
-							[:eachClass |
-							messages addAll: eachClass selectors.
-							messages size > anInteger ifTrue: [^#()]]]
-						ifFalse: 
-							[
-							[:eachClass |
-							eachClass selectors do: 
-									[:eachSelector |
-									(eachSelector beginsWith: aString ignoreCase: ignoreCase)
-										ifTrue: 
-											[messages add: eachSelector.
-											messages size > anInteger ifTrue: [^#()]]]]]).
-			messages])
-			collect: [:each | ScintillaListItem text: each icon: icon]!
+	class := self classForToken: anAssociation.
+	class == UndefinedObject ifTrue: [class := nil].
```

Comment:
Why would you want to treat this as a special case? If I type nil in a workspace, I'd expect to be offered only messages understood by UndefinedObject in the auto-completion list.